src/sentry/buffer/redis.py

```diff
@@ -172,20 +175,23 @@ def incr(self, model, columns, filters, extra=None, signal_only=None, return_inc
         - Add hashmap key to pending flushes
         """
 
-        # TODO(dcramer): longer term we'd rather not have to serialize values
-        # here (unless it's to JSON)
         key = self._make_key(model, filters)
         pending_key = self._make_pending_key_from_key(key)
         # We can't use conn.map() due to wanting to support multiple pending
         # keys (one per Redis partition)
-        conn = self.cluster.get_local_client_for_key(key)
+        if self.is_redis_cluster:
+            conn = self.cluster
+        else:
+            conn = self.cluster.get_local_client_for_key(key)
 
         pipe = conn.pipeline()
         pipe.hsetnx(key, "m", f"{model.__module__}.{model.__name__}")
-        # TODO(dcramer): once this goes live in production, we can kill the pickle path
-        # (this is to ensure a zero downtime deploy where we can transition event processing)
-        pipe.hsetnx(key, "f", pickle.dumps(filters))
-        # pipe.hsetnx(key, 'f', json.dumps(self._dump_values(filters)))
+
+        if self.is_redis_cluster:
+            pipe.hsetnx(key, "f", json.dumps(self._dump_values(filters)))
```

Comment:
Let's figure out a path here to validate the shape of the JSON data even for the non JSON path so we know that we have a path forward.